### PR TITLE
Extract layers module from rpc crate

### LINKED ISSUE #8161: Extract layers module from rpc crate
### Describe the feature

These layers aren't used by the crate directly, only in rpc-builder, they don't even need any eth/reth types

https://github.com/paradigmxyz/reth/tree/bdb8238d79c383033a68cbeef8809c37a481e303/crates/rpc/rpc/src/layers

## TODO
* extract these types to a standalone crate `rpc/rpc-layer` or `rpc/rpc-util`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rupam-04**
Closes #8161

### CODE REVIEW

**Comment by rupam-04** [comment]
@mattsse any updates on this?

File: Cargo.toml [hunk 1]
```diff
@@ -90,7 +90,7 @@ members = [
     "examples/exex/*",
     "examples/db-access",
     "testing/ef-tests/",
-    "testing/testing-utils",
+    "testing/testing-utils", "crates/rpc/rpc-layer",
```
Comment (emhane) [inline Cargo.toml:93]:
```suggestion
    "testing/testing-utils",
    "crates/rpc/rpc-layer",
```
Nitpick

**Review by emhane — APPROVED** [review]
lgtm

**Comment by rupam-04** [comment]
@emhane any idea why some checks are failing?

**Comment by emhane** [comment]
try locally 
```
rm Cargo.lock
cargo clean
cargo c
```
then commit the updated lock file, and push. that should do it. @rupam-04

**Comment by rupam-04** [comment]
> try locally
> 
> ```
> rm Cargo.lock
> cargo clean
> cargo c
> ```
> 
> then commit the updated lock file, and push. that should do it. @rupam-04

worked, ty

File: Cargo.toml [hunk 2]
```diff
@@ -47,7 +47,6 @@ members = [
     "crates/rpc/rpc-testing-util/",
     "crates/rpc/rpc-types/",
     "crates/rpc/rpc-types-compat/",
-    "crates/rpc/rpc-layer",
```
Comment (emhane) [inline Cargo.toml]: we like this alphabetically sorted, please put it back here and remove it from down below

**Review by emhane — CHANGES_REQUESTED** [review]

**Review by emhane — APPROVED** [review]

**Pull request merged**